flow/probes/pcap.go

```diff
@@ -0,0 +1,399 @@
+			//				FlowDataLength uint32
+		},
+		HeaderProtocol: layers.SFlowProtoEthernet,
+		//	FrameLength    uint32
+		PayloadRemoved: 0,
+		//	HeaderLength   uint32
+		//	Header         gopacket.NewPacket
+	}
+
+	sflowSeq++
+
+	sf := layers.SFlowFlowSample{
+		EnterpriseID: layers.SFlowStandard,
+		Format:       layers.SFlowTypeFlowSample,
+		//	SampleLength    uint32
+		SequenceNumber:  uint32(sflowSeq),
+		SourceIDClass:   layers.SFlowTypeSingleInterface,
+		SourceIDIndex:   layers.SFlowSourceValue(47),
+		SamplingRate:    300,
+		SamplePool:      0x12345,
+		Dropped:         0,
+		InputInterface:  48,
+		OutputInterface: 47,
+		//		RecordCount:     1,
+		//		Records:
+	}
+	sf.Records = append(sf.Records, sfraw)
+	sf.RecordCount = uint32(len(sf.Records))
+
+	sflowLayer := &layers.SFlowDatagram{
+		DatagramVersion: 5,
+		AgentAddress:    net.IP{127, 0, 0, 3},
+		SubAgentID:      0,
+		SequenceNumber:  sflowSeq,
+		AgentUptime:     2294190,
+		//		SampleCount:     count,
+		//		FlowSamples:     sflowsamples,
+		//				CounterSamples  []SFlowCounterSample
+	}
+	sflowLayer.FlowSamples = append(sflowLayer.FlowSamples, sf)
+	sflowLayer.SampleCount = uint32(len(sflowLayer.FlowSamples))
+
+	rawBytes := SFlowDatagramSerialize(sflowLayer, packets)
+	return rawBytes
+}
+
+func genEthIPUdp(payload []byte) []byte {
+	ethernetLayer := &layers.Ethernet{
+		SrcMAC: net.HardwareAddr{0x00, 0x01, 0xFF, 0xAA, 0xFA, 0xAA},
+		DstMAC: net.HardwareAddr{0x00, 0x01, 0xBD, 0xBD, 0xBD, 0xBD},
+	}
+	ethernetLayer.EthernetType = layers.EthernetTypeIPv4
+	ipLayer := &layers.IPv4{
+		SrcIP: net.IP{127, 0, 0, 1},
+		DstIP: net.IP{127, 0, 0, 2},
+	}
+	ipLayer.IHL = 0x45
+	ipLayer.Protocol = layers.IPProtocolUDP
+	udpLayer := &layers.UDP{
+		SrcPort: layers.UDPPort(54321),
+		DstPort: layers.UDPPort(6343),
+	}
+	udpLayer.Length = uint16(8 + len(payload))
+	ipLayer.Length = 20 + udpLayer.Length
+	ipLayer.TTL = 64
+	ipLayer.Id = uint16(0xbbea)
+
+	// And create the packet with the layers
+	buffer := gopacket.NewSerializeBuffer()
+	options := gopacket.SerializeOptions{}
+	gopacket.SerializeLayers(buffer, options,
+		ethernetLayer,
+		ipLayer,
+		udpLayer,
+		gopacket.Payload(payload),
+	)
+
+	return buffer.Bytes()
+}
+
+func writePcap(packet []byte) {
+	logging.GetLogger().Debug("Writing PCAP file")
+	f, _ := os.Create("/tmp/file.pcap")
+	w := pcapgo.NewWriter(f)
+	w.WriteFileHeader(65536, layers.LinkTypeEthernet) // new file, must do this.
+	w.WritePacket(
+		gopacket.CaptureInfo{Timestamp: time.Now(),
+			CaptureLength: len(packet),
+			Length:        len(packet),
+		}, packet)
+	f.Close()
+}
+
+func (probe *PcapProbe) AsyncProgressInfo() {
+	ticker := time.NewTicker(10 * time.Second)
+	defer func() {
+		ticker.Stop()
+	}()
+	for {
+		select {
+		case <-ticker.C:
+			logging.GetLogger().Debug("%d packets replayed | %s", nbpackets, probe.flowtable.String())
+		}
+	}
+}
+
+func (probe *PcapProbe) Start() error {
+	f, err := os.Open(probe.Filename)
```

Comment:
This probe should live capture from an interface, maybe specified from a conf parameter for a first step. As specified before I would prefer to have a other tool in order to read pcap file a send sflow packet to the agent.